pkg/process/metadata/parser/service.go

```diff
@@ -285,6 +287,15 @@ func parseCommandContextJava(args []string) string {
 
 			if arg = trimColonRight(arg); isRuneLetterAt(arg, 0) {
 				if strings.HasSuffix(arg, javaJarExtension) {
+					// try to introspect the jar to get service name from spring application name
+					// TODO: pass process envs
+					springAppName, err := javaparser.GetSpringBootAppName(process.Cwd, a, args)
+					if err != nil {
+						log.Tracef("Error while trying to extract properties from potential spring boot application: %v", err)
+					} else if len(springAppName) > 0 {
+						return springAppName
+					}
+					// return the jar
```

Comment:
I do want to get one more algorithm in place here, too, but maybe that can be another PR. It'd be good to check if it's a Tomcat, WebSphere, or WebLogic. If it is, we can look for the .war files and extract servlet context information from web.xml. Barring that, we can just take the names the .war files. Then we can merge the names of all the services into one to make a single service name for outside of the app.

Reply:
Yes of course this is intended to be in separate PR(s) since the code will be way different so I preferred to split. The web.xml does not sound to me the right place to look at since the root context mapping is vendor dependent. A general approach can be to use the war name. Otherwise we should look at more specific places (can be the server.xml for tomcat, the weblogic.xml packaged in the war for weblogic, etc.). The simplest is to start with the war name.